Opencast 17 doesn't start with CAS config enabled

### Discussed in https://github.com/orgs/opencast/discussions/6474

<div type='discussions-op-text'>

<sup>Originally posted by **Meridiana** February  6, 2025</sup>
Hello,

Opencast 17 doesn't start if CAS authentication is configured (16.7 does with same config):

```
2025-02-06T12:25:42,124 | ERROR | (BootFeaturesInstaller:122) - Error installing boot features
org.apache.felix.resolver.reason.ReasonException: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=opencast-security-cas; type=karaf.feature; version="[17.0.0,17.0.0]"; filter:="(&(osgi.identity=opencast-security-cas)(type=karaf.feature)(version>=17.0.0)(version<=17.0.0))" [caused by: Unable to resolve opencast-security-cas/17.0.0: missing requirement [opencast-security-cas/17.0.0] osgi.identity; osgi.identity=org.springframework.security.cas; type=osgi.bundle; version="[3.1.7.RELEASE,3.1.7.RELEASE]"; resolution:=mandatory [caused by: Unable to resolve org.springframework.security.cas/3.1.7.RELEASE: missing requirement [org.springframework.security.cas/3.1.7.RELEASE] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.jasig.cas.client.authentication)(version>=3.1.1)(!(version>=3.2.0)))" [caused by: Unable to resolve opencast-security-cas-client-wrapper/17.0.0: missing requirement [opencast-security-cas-client-wrapper/17.0.0] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.apache.xml.security)(version>=4.0.0)(!(version>=5.0.0)))" [caused by: Unable to resolve org.apache.santuario.xmlsec/4.0.2: missing requirement [org.apache.santuario.xmlsec/4.0.2] osgi.wiring.package; filter:="(&(osgi.wiring.package=jakarta.xml.bind)(version>=4.0.0)(!(version>=5.0.0)))"]]]]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[?:?]
        at org.apache.felix.resolver.ResolverImpl.doResolve(ResolverImpl.java:433) ~[?:?]
        at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:420) ~[?:?]
        at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:374) ~[?:?]
        at org.apache.karaf.features.internal.region.SubsystemResolver.resolve(SubsystemResolver.java:256) ~[?:?]
        at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:399) ~[?:?]
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1069) ~[?:?]
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$13(FeaturesServiceImpl.java:1004) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:840) [?:?]
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve opencast-security-cas/17.0.0: missing requirement [opencast-security-cas/17.0.0] osgi.identity; osgi.identity=org.springframework.security.cas; type=osgi.bundle; version="[3.1.7.RELEASE,3.1.7.RELEASE]"; resolution:=mandatory [caused by: Unable to resolve org.springframework.security.cas/3.1.7.RELEASE: missing requirement [org.springframework.security.cas/3.1.7.RELEASE] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.jasig.cas.client.authentication)(version>=3.1.1)(!(version>=3.2.0)))" [caused by: Unable to resolve opencast-security-cas-client-wrapper/17.0.0: missing requirement [opencast-security-cas-client-wrapper/17.0.0] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.apache.xml.security)(version>=4.0.0)(!(version>=5.0.0)))" [caused by: Unable to resolve org.apache.santuario.xmlsec/4.0.2: missing requirement [org.apache.santuario.xmlsec/4.0.2] osgi.wiring.package; filter:="(&(osgi.wiring.package=jakarta.xml.bind)(version>=4.0.0)(!(version>=5.0.0)))"]]]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[?:?]
        ... 12 more
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve org.springframework.security.cas/3.1.7.RELEASE: missing requirement [org.springframework.security.cas/3.1.7.RELEASE] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.jasig.cas.client.authentication)(version>=3.1.1)(!(version>=3.2.0)))" [caused by: Unable to resolve opencast-security-cas-client-wrapper/17.0.0: missing requirement [opencast-security-cas-client-wrapper/17.0.0] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.apache.xml.security)(version>=4.0.0)(!(version>=5.0.0)))" [caused by: Unable to resolve org.apache.santuario.xmlsec/4.0.2: missing requirement [org.apache.santuario.xmlsec/4.0.2] osgi.wiring.package; filter:="(&(osgi.wiring.package=jakarta.xml.bind)(version>=4.0.0)(!(version>=5.0.0)))"]]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[?:?]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[?:?]
        ... 12 more
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve opencast-security-cas-client-wrapper/17.0.0: missing requirement [opencast-security-cas-client-wrapper/17.0.0] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.apache.xml.security)(version>=4.0.0)(!(version>=5.0.0)))" [caused by: Unable to resolve org.apache.santuario.xmlsec/4.0.2: missing requirement [org.apache.santuario.xmlsec/4.0.2] osgi.wiring.package; filter:="(&(osgi.wiring.package=jakarta.xml.bind)(version>=4.0.0)(!(version>=5.0.0)))"]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[?:?]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[?:?]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[?:?]
        ... 12 more
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve org.apache.santuario.xmlsec/4.0.2: missing requirement [org.apache.santuario.xmlsec/4.0.2] osgi.wiring.package; filter:="(&(osgi.wiring.package=jakarta.xml.bind)(version>=4.0.0)(!(version>=5.0.0)))"
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[?:?]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[?:?]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[?:?]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[?:?]
        ... 12 more
```

It is an AllInOne installation on Ubuntu 24.04 ...

Regards,
Meridiana

*EDIT(@lkiesow): formatted the error logs*

--- a/pom.xml
+++ b/pom.xml
@@ -66,7 +66,7 @@
     <rest-assured.version>4.4.0</rest-assured.version>
     <spring.version>3.1.4.RELEASE</spring.version>
     <spring-security.version>3.1.7.RELEASE</spring-security.version>
-    <xmlsec.version>4.0.2</xmlsec.version>
+    <xmlsec.version>2.2.6</xmlsec.version>
   </properties>
   <modules>
     <module>modules/admin-ui</module>

